NorthstarDedicatedTest/serverauthentication.cpp

```diff
@@ -516,7 +518,32 @@ bool ProcessConnectionlessPacketHook(void* a1, netpacket_t* packet)
 	return ProcessConnectionlessPacket(a1, packet);
 }
 
-void CServerGameDLL__OnRecievedSayTextMessageHook(void* self, unsigned int senderClientIndex, const char* message, char unknown)
+void ReplaceStringInPlace(std::string& subject, const std::string& search, const std::string& replace)
+{
+	size_t pos = 0;
+	while ((pos = subject.find(search, pos)) != std::string::npos)
+	{
+		subject.replace(pos, search.length(), replace);
+		pos += replace.length();
+	}
+}
+
+std::string currentMessage;
+int currentPlayerId;
+int currentChannelId;
+bool shouldBlock;
+bool isProcessed = true;
+
+SQRESULT setMessage(void* sqvm)
+{
+	currentMessage = ServerSq_getstring(sqvm, 1);
+	currentPlayerId = ServerSq_getinteger(sqvm, 2);
+	currentChannelId = ServerSq_getinteger(sqvm, 3);
+	shouldBlock = ServerSq_getbool(sqvm, 4);
+	return SQRESULT_NOTNULL;
+}
+
+void CServerGameDLL__OnRecievedSayTextMessageHook(void* self, unsigned int senderClientIndex, const char* message, int channelId)
```

Comment:
Can you fix all occurrences of this _received_ typo please?

Reply:
i before e except after c smh